### ISSUE TITLE
[2020 ICML] A Simple Framework for Contrastive Learning of Visual Representations

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#69]
SimCLR v1 논문

기존의 방법에서 다양한 실험들을 통해 다음을 보임

1. Composition of data augmentations plays a critical role in defining effective predictive tasks
2. Learnable nonlinear transformation between the representations and the contrastive loss substantially improves the quality of the learned representations
3. Contrastive learning benefits from larger batch sizes and more training steps compared to supervised learning

요약하면 다음과 같음
1. Data augmentation을 복합적으로 잘 사용하면 성능이 개선됨
2. MLP layer를 쌓아서 학습하면 성능이 개선됨
3. Large batch size를 사용하고 training step을 길게하면 성능이 개선됨

### 1.1. The Contrastive Learning Framework
<img width="649" alt="image" src="https://user-images.githubusercontent.com/49643709/168644525-7362e586-e122-4f3e-bd40-26b1b45847f5.png">
<img width="646" alt="image" src="https://user-images.githubusercontent.com/49643709/168644802-00df57b1-98c1-42d7-ad43-0277374cefaf.png">
<img width="629" alt="image" src="https://user-images.githubusercontent.com/49643709/168645236-06ea64d6-d3e3-4570-9041-71f392eea548.png">

해당 loss는 NT-Xent loss (normalized temperature-scaled cross entropy loss)라고 명칭

Loss에 indicator function을 넣어주는 이유는 N개의 batch에 대하여 자신을 제외한 2N-1에 대하여 contrastive learning을 할 것이기 때문임
(1장의 이미지 - 1개의 positive, 2N-2개의 negative samples에 대하여 contrastive learning)

### 1.2. Training with Larger Batch Size

기존의 SGD/Momentum optimizer with linear learning rate scaling으로 large batch를 학습하게 되면 unstable함

Stabilize를 위해 모든 batch size에 대해 LARS optimizer 사용

Model이 local information leakage를 학습하지 않고 representation을 학습하도록 모든 device에서 BN mean, variance를 aggregate

### 1.3. Evaluation Protocol

Linear evaluation protocol을 이용하여 성능 측정

Default setting은 다음과 같음
- Data augmentation
    - RandomResizedCrop(size=(224, 224), scale=(0.08, 1.0), ratio=(3/4, 4/3))
    - RandomHorizontalFlip(p=0.5)
    - RandomApply(ColorJitter(brightness=0.8, contrast=0.8, saturation=0.8, hue=0.2), p=0.8)
    - RandomGrayscale(p=0.2)
    - RandomApply(Gaussian blur(kernel_size=(22, 22), sigma=(0.1, 2.0)), p=0.5)
- Model
    - ResNet-50
